spaces in toolchain linker path no longer working

### Describe the bug

After the change in commit d77b58aacb913d25c30b63620d413eaa94622750 our tool-chain is no longer working. 

We are using the GNU Arm Embedded on Windows installed in the default install directory which contains spaces. (C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi). 
This used to work fine but after updating to the latest main the following linker error appeared:

`arm-none-eabi/bin/ld.bfd.exe: cannot find Files: No such file or directory`
`arm-none-eabi/bin/ld.bfd.exe: cannot find (x86)/Arm: No such file or directory`
`arm-none-eabi/bin/ld.bfd.exe: cannot find GNU: No such file or directory`
`arm-none-eabi/bin/ld.bfd.exe: cannot find Toolchain: No such file or directory`
`arm-none-eabi/bin/ld.bfd.exe: cannot find arm-none-eabi/14.3: No such file or directory`
`arm-none-eabi/bin/ld.bfd.exe: cannot find rel1/bin/../lib/gcc/arm-none-eabi/14.3.1/thumb/v7e-m/nofp: No such file or directory`

I have not spotted the root cause yet but my guess is that due to the late computation of the linker path some string-replace has been missed. 
when re-installing the gnu tool-chain to a path without spaces and adjusting the environment variable the build is working again 

### Regression

- [x] This is a regression.

### Steps to reproduce

install the gnu tool-chain in the default location (with spaces) and set the environment variables such as described here 
https://docs.zephyrproject.org/latest/develop/toolchains/gnu_arm_embedded.html 
compile any project using a board with an arm processor

### Relevant log output

the toolchain path is still found and set


```shell
-- Found toolchain: gnuarmemb (C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\14.3 rel1)
... 
-- Found GnuLd: C:/Program Files (x86)/Arm GNU Toolchain arm-none-eabi/14.3 rel1/arm-none-eabi/bin/ld.bfd.exe (found version "2.44.0")
-- The C compiler identification is GNU 14.3.1
-- The CXX compiler identification is GNU 14.3.1
-- The ASM compiler identification is GNU
-- Found assembler: C:/Program Files (x86)/Arm GNU Toolchain arm-none-eabi/14.3 rel1/bin/arm-none-eabi-gcc.exe
```
but the linker command no longer works

```shell
C:/PROGRA~2/ARMGNU~1/142BCE~1.3RE/bin/../lib/gcc/arm-none-eabi/14.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: cannot find Files: No such file or directory
C:/PROGRA~2/ARMGNU~1/142BCE~1.3RE/bin/../lib/gcc/arm-none-eabi/14.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: cannot find (x86)/Arm: No such file or directory
C:/PROGRA~2/ARMGNU~1/142BCE~1.3RE/bin/../lib/gcc/arm-none-eabi/14.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: cannot find GNU: No such file or directory
C:/PROGRA~2/ARMGNU~1/142BCE~1.3RE/bin/../lib/gcc/arm-none-eabi/14.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: cannot find Toolchain: No such file or directory
C:/PROGRA~2/ARMGNU~1/142BCE~1.3RE/bin/../lib/gcc/arm-none-eabi/14.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: cannot find arm-none-eabi/14.3: No such file or directory
C:/PROGRA~2/ARMGNU~1/142BCE~1.3RE/bin/../lib/gcc/arm-none-eabi/14.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: cannot find rel1/bin/../lib/gcc/arm-none-eabi/14.3.1/thumb/v7e-m/nofp: No such file or directory
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS Windows
- toolchain: gnuarmemb 

### Additional Context

_No response_

--- a/cmake/compiler/target_template.cmake
+++ b/cmake/compiler/target_template.cmake
@@ -102,7 +102,7 @@ if(NOT COMMAND compiler_set_linker_properties)
     # Compute the library directory name
 
     get_filename_component(library_dir ${library_path} DIRECTORY)
-    set_linker_property(PROPERTY lib_include_dir "-L${library_dir}")
+    set_linker_property(PROPERTY lib_include_dir "-L\"${library_dir}\"")
 
     # Compute the linker option for this library
 

